sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -1251,6 +1294,16 @@ private void ensureFromNotCalledYet() {
           getJsonTableRef() == null && getQuery() == null, "from() or fromQuery() already called");
     }
 
+    private void ensureReadOptionsNotSet() {
+      checkState(getReadOptions() == null, "withReadOptions() already called");
+    }
+
+    private void ensureSelectedFieldsAndRowRestrictionNotSet() {
```

Comment:
For a little future-proofing in case additional read options are added to the BQ Storage API in the future, these methods could be named `ensureReadOptionsObjectNotSet` and `ensureIndividualReadOptionsNotSet`.

Reply:
Done.